QRCoder/QRCodeGenerator.cs

```diff
@@ -1134,16 +1224,27 @@ private static string PlainTextToBinaryByte(string plainText, EciMode eciMode, b
                 }
             }
 
-            foreach (var b in codeBytes)
-                codeText += DecToBin(b, 8);
-
-            return codeText;
+            return ToBitArray(codeBytes);
         }
 
+        private static BitArray ToBitArray(byte[] byteArray, int prefixZeros = 0)
+        {
+            // new BitArray(byteArray) is not used because it reverses the bit order within each byte
+            var bitArray = new BitArray(byteArray.Length * 8 + prefixZeros);
+            for (var i = 0; i < byteArray.Length; i++)
+            {
+                var byteVal = byteArray[i];
+                for (var j = 0; j < 8; j++)
+                {
+                    bitArray[i * 8 + j + prefixZeros] = (byteVal & (1 << (7 - j))) != 0;
```

Comment:
```suggestion
                    bitArray[(int)((uint)i * 8) + j + prefixZeros] = (byteVal & (1 << (7 - j))) != 0;
```
Or use the bit-hack directly, to avoid the casts which make the code more unreadable. But w/ the bit-hack maybe a comment is needed?!